From 9a2404c2cbb6550f772bd6b72bc6ee1950cf0043 Mon Sep 17 00:00:00 2001
From: Loc Vu <loc.vu.zn@renesas.com>
Date: Thu, 11 May 2017 23:26:11 +0700
Subject: [PATCH 023/433] ARM: shmobile: r8a7743: Allow booting secondary CPU
 cores in debug mode

allow booting secondary CPU cores in debug mode

Signed-off-by: Binh Nguyen <binh.nguyen.uw@rvc.renesas.com>
Signed-off-by: Loc Vu <loc.vu.zn@renesas.com>
---
 arch/arm/mach-shmobile/smp-r8a7743.c | 14 +-------------
 1 file changed, 1 insertion(+), 13 deletions(-)

diff --git a/arch/arm/mach-shmobile/smp-r8a7743.c b/arch/arm/mach-shmobile/smp-r8a7743.c
index 09b1e76..3cf4452 100644
--- a/arch/arm/mach-shmobile/smp-r8a7743.c
+++ b/arch/arm/mach-shmobile/smp-r8a7743.c
@@ -42,21 +42,9 @@ static void __init r8a7743_smp_prepare_cpus(unsigned int max_cpus)
 	rcar_gen2_pm_init();
 }
 
-static int r8a7743_smp_boot_secondary(unsigned int cpu,
-				      struct task_struct *idle)
-{
-	/* Error out when hardware debug mode is enabled */
-	if (rcar_gen2_read_mode_pins() & BIT(21)) {
-		pr_warn("Unable to boot CPU%u when MD21 is set\n", cpu);
-		return -ENOTSUPP;
-	}
-
-	return shmobile_smp_apmu_boot_secondary(cpu, idle);
-}
-
 struct smp_operations r8a7743_smp_ops __initdata = {
 	.smp_prepare_cpus	= r8a7743_smp_prepare_cpus,
-	.smp_boot_secondary	= r8a7743_smp_boot_secondary,
+	.smp_boot_secondary	= shmobile_smp_apmu_boot_secondary,
 #ifdef CONFIG_HOTPLUG_CPU
 	.cpu_can_disable	= shmobile_smp_cpu_can_disable,
 	.cpu_die		= shmobile_smp_apmu_cpu_die,
-- 
2.7.4

